### PR TITLE
adds auth middleware on all calls to /api/schema

### PR DESCRIPTION
**Pull request by owen-d**

### CODE REVIEW

File: server/users/userController.js [hunk 1]
```diff
@@ -82,9 +82,14 @@ module.exports = {
   },
 
   checkAuth: function(request, response, next) {
-    //TODO: still need communication from client side to test sent headers (#91)
-    //this will be whatever header we put the jwt under.
-    var token = request.headers["x-jwt"];
-    var user = jwt.verify(token, process.env.JWT_SECRET);
+    var token = request.headers['x-jwt'];
+    jwt.verify(token, process.env.JWT_SECRET, function(err, decoded) {
```
Comment (dduugg) [inline server/users/userController.js]: pretty sure where using 'error' everywhere else in the server code

File: server/users/userController.js [hunk 2]
```diff
@@ -82,9 +82,14 @@ module.exports = {
   },
 
   checkAuth: function(request, response, next) {
-    //TODO: still need communication from client side to test sent headers (#91)
```
Comment (dduugg) [inline server/users/userController.js]: Is this resolved?

File: server/users/userController.js (hunk 1 above)
Comment (owen-d) [inline server/users/userController.js]: pretty sure we're using 'error' everywhere else in the server code

**Comment by owen-d** [comment]
Resolves #91

File: server/users/userController.js [hunk 3]
```diff
@@ -82,9 +82,14 @@ module.exports = {
...
-    //this will be whatever header we put the jwt under.
     var token = request.headers["x-jwt"];
-    var user = jwt.verify(token, process.env.JWT_SECRET);
+    jwt.verify(token, process.env.JWT_SECRET, function(err, decoded) {
```
Comment (dduugg) [inline server/users/userController.js]: use 'error'

**Pull request merged**